core/capabilities/webapi/trigger.go

```diff
@@ -1,18 +1,310 @@
-	// TODO (CAPPL-22, CAPPL-24):
-	//   - decode config
-	//   - create an implementation of the capability API and add it to the Registry
-	//   - create a handler and register it with Gateway Connector
-	//   - manage trigger subscriptions
-	//   - process incoming trigger events and related metadata
-	return nil, nil
+type Input struct {
+}
+type TriggerConfig struct {
+	AllowedSenders []string `toml:"allowedSenders"`
+	AllowedTopics  []string `toml:"allowedTopics"`
+	// RateLimiter    common.RateLimiterConfig `toml:"rateLimiter"`
+	// RateLimiter    *values.Map `toml:"rateLimiter"`
+	RequiredParams []string `toml:"requiredParams"`
+}
+
+// TODO: Question asked in Web API Trigger about this structure
+// https://docs.google.com/document/d/1mCTAo-ix-P923eUlh4SloZfBN9PCvgf90oHWbmykjsc/edit?disco=AAABWF65hAM
+// Eventually can have PENDING or COMPLETED so status is different than Success
+
+type Response struct {
+	Success      bool   `json:"success"`
+	ErrorMessage string `json:"error_message,omitempty"`
+	Status       string `json:"status"`
+}
+
+type webapiTrigger struct {
+	allowedSendersMap map[string]bool
+	allowedTopicsMap  map[string]bool
+	ch                chan<- capabilities.TriggerResponse
+	config            TriggerConfig
+	// rateLimiter       *common.RateLimiter
+}
+
+type triggerConnectorHandler struct {
+	services.StateMachine
+
+	capabilities.CapabilityInfo
+	capabilities.Validator[TriggerConfig, Input, capabilities.TriggerResponse]
+	connector           connector.GatewayConnector
+	lggr                logger.Logger
+	mu                  sync.Mutex
+	registeredWorkflows map[string]webapiTrigger
+	signerKey           *ecdsa.PrivateKey
+}
+
+var _ capabilities.TriggerCapability = (*triggerConnectorHandler)(nil)
+var _ services.Service = &triggerConnectorHandler{}
+
+func NewTrigger(config string, registry core.CapabilitiesRegistry, connector connector.GatewayConnector, signerKey *ecdsa.PrivateKey, lggr logger.Logger) (*triggerConnectorHandler, error) {
+	handler := &triggerConnectorHandler{
+		Validator:           capabilities.NewValidator[TriggerConfig, Input, capabilities.TriggerResponse](capabilities.ValidatorArgs{Info: webapiTriggerInfo}),
+		connector:           connector,
+		signerKey:           signerKey,
+		registeredWorkflows: map[string]webapiTrigger{},
+		lggr:                lggr.Named("WorkflowConnectorHandler"),
+	}
+
+	return handler, nil
+}
+
+// Iterate over each topic, checking against senders and rateLimits, then starting event processing and responding
+func (h *triggerConnectorHandler) processTrigger(ctx context.Context, gatewayID string, body *api.MessageBody, sender ethCommon.Address, payload workflow.TriggerRequestPayload) {
+	wrappedPayload, _ := values.WrapMap(payload)
+
+	for _, trigger := range h.registeredWorkflows {
+		topics := payload.Topics
+
+		for _, topic := range topics {
```

Comment:
This logic makes it so that only an exact match on topics of a workflow will cause an execution. Is this the design that we want? We should define topics